### ISSUE TITLE
Use the current primary instance disk size when creating replicas

### ISSUE THREAD

**Issue by jawnsy** [terraform-google-modules/terraform-google-sql-db#338]
### TL;DR

When creating replicas, it is mandatory to specify the disk size, but that may have grown since the primary instance was created. When dynamically creating and destroying replicas, such as for test purposes, the primary may have grown in size since the initial creation, leading to errors.

### Detailed design

It may be possible to read the current instance disk size from the data object; I'm unsure here. If it is not possible, then implementing an experience like the one described below may not be possible.

### Additional information

Consider the following scenario:

1. Terraform configuration includes a primary database and replica, both sized to 10 GB, with automatic disk resize enabled
2. A workload continuously loads the database, expanding it to 100 GB
3. If we destroy the replica and re-create it, then the same Terraform code that initially worked, no longer works

Since it is not possible to have a replica with a smaller disk size than the primary, and it is not generally useful to have a read-only replica with a _larger_ disk size than the primary, it seems reasonable to default to the current disk size of the primary when creating the replica resource, at creation time.

It is not possible to implement this externally, because the postgresql submodule creates both the primary and the replica

**Comment by jawnsy** [comment]
I believe this issue is still valid

**Comment by jawnsy** [comment]
I believe this issue is still valid

**Comment by jawnsy** [comment]
I believe this issue is still valid

**Comment by jawnsy** [comment]
This issue is still valid

**Comment by jawnsy** [comment]
This issue is still valid
